### PR TITLE
Document nix version pinning more explicitly

### PR DESCRIPTION
**Pull request by kolloch**
According to https://github.com/DeterminateSystems/nix-installer/issues/387#issuecomment-1581743074

##### Description

Document nix-installer and nix version pinning more explicitly.

My editor also removed some trailing space in the README.md automatically -- I hope that's fine, otherwise I'll clean that up.

##### Checklist

(doc only)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -412,6 +412,19 @@ curl -sSf -L https://github.com/DeterminateSystems/nix-installer/releases/downlo
 ./nix-installer install
 ```
 
+Each installer version has an [associated pre ferred hard-coded nix
```
Comment (Hoverbear) [inline README.md:415]:
```suggestion
Each installer version has an [associated preferred hard-coded nix
```

File: README.md [hunk 2]
```diff
@@ -412,6 +412,19 @@ curl -sSf -L https://github.com/DeterminateSystems/nix-installer/releases/downlo
...
+version](src/settings.rs) - if you pin the installer version, you'll also
+indirectly pin to the associated nix version.
+
+## Pin nix version directly
```
Comment (Hoverbear) [inline README.md:419]:
```suggestion

```

**Review by Hoverbear — COMMENTED** [review]
A couple comments, otherwise looks good, thank you!

**Comment by Nebucatnetzer** [comment]
During a POC setup we noticed that we ended up with different Nix versions.
IMO this should be already mentioned in the getting started section, especially considering were talking about a Nix related tool.

**Comment by cole-h** [comment]
Thanks for your contribution, and sorry for the delay! I bundled the contents of this PR into https://github.com/DeterminateSystems/nix-installer/pull/854 with some minor adjustments (because of some CI issues caused by misconfiguration for external contributors, CI would never have succeeded on your PR; we'll get that squared away in the near future).

**Pull request closed without merging**